sandy/core/cov.py

```diff
@@ -687,15 +776,29 @@ def _gls_G(self, S, Vy_extra):
                     	0	      1
         0	6.00000e+00	1.10000e+01
         1	1.10000e+01	2.60000e+01
+
+        >>> S = np.array([[1, 2], [3, 4]])
+        >>> sensitivity = sandy.CategoryCov.from_var([1, 1])
+        >>> Vy = np.diag(pd.Series([1, 1]))
+        >>> sensitivity._gls_G(S, Vy, sparse=True)
+                    	0	      1
+        0	6.00000e+00	1.10000e+01
+        1	1.10000e+01	2.60000e+01
         """
         index = pd.DataFrame(Vy_extra).index  # Simetry of cov matrix
         Vy_extra_ = sandy.CategoryCov(Vy_extra).data.values
         # GLS_sensitivity:
         Vy_calc = self._gls_Vy_calc(S).values
+        if sparse is True:
+            Vy_calc = sps.csr_matrix(Vy_calc)

```

Comment:
Thos is not really necessary, right?!

Reply:
I think that all mathematical operation are more optimized for sparse matrices, because here only sum non null element. I numpy i thouht that sum all the elements, so we are avoiding a lot of operations

Reply:
Ok, but then, can you use only 1 if/else clause instead of 2? To be consistent everywhere in the module (sometimes you use 1 and other times 2)